pallets/identity/src/auth.rs

```diff
@@ -186,8 +189,22 @@ impl<T: Config> Module<T> {
             ensure!(expiry > now, AuthorizationError::Expired);
         }
 
+        if auth.count == 0 {
+            <Authorizations<T>>::remove(&target, auth_id);
+            <AuthorizationsGiven<T>>::remove(auth.authorized_by, auth_id);
+        }
+
         // Run custom per-type validation and updates.
-        accepter(auth.authorization_data, auth.authorized_by)?;
+        let res = accepter(auth.authorization_data, auth.authorized_by);
+
+        if res.is_err() {
+            // decrement and  update authorization
+            <Authorizations<T>>::mutate(&target, auth_id, |auth| {
+                auth.as_mut().unwrap().count -= 1;
+            });
```

Comment:
Never use `unwrap()` or anything else that can panic.  That would abort the whole block production and cause problems with the chain.

You don't need to use `<Authorizations<T>>::mutate()`, since you already have a copy of `auth` that you can modify then store with `<Authorizations<T>>::insert()` to replace the old value.